React/Views/RCTNavItem.m

```diff
@@ -15,6 +15,13 @@ @implementation RCTNavItem
 @synthesize leftButtonItem = _leftButtonItem;
 @synthesize rightButtonItem = _rightButtonItem;
 
+-(id)init {
+  if (self = [super init]) {
+    _rightButtonSystemItem = -1;
```

Comment:
Can you use NSNotFound instead of -1 as the "null" value? It's more idiomatic for Obj-C (and also doesn't rely on the implementation detail that UIBarButtonSystemItem happens to be an NSInteger rather than NSUInteger).